sentry-android-fragment/src/main/java/io/sentry/android/fragment/SentryFragmentLifecycleCallbacks.kt

```diff
@@ -91,4 +117,45 @@ class SentryFragmentLifecycleCallbacks(
     private fun getFragmentName(fragment: Fragment): String {
         return fragment.javaClass.simpleName
     }
+
+    private fun isRunningSpan(fragment: Fragment): Boolean =
+        fragmentsWithOngoingTransactions.containsKey(fragment)
+
+    private fun startTracing(fragment: Fragment) {
+        if (!isPerformanceEnabled || isRunningSpan(fragment)) {
+            return
+        }
+
+        var transaction: ISpan? = null
+        hub.configureScope {
+            transaction = it.transaction
+        }
+
+        val fragmentName = getFragmentName(fragment)
+        val span = transaction?.startChild(FRAGMENT_LOAD_OP, fragmentName)
+
+        span?.let {
+            fragmentsWithOngoingTransactions[fragment] = it
+        }
+    }
+
+    private fun stopTracing(fragment: Fragment) {
+        if (!isPerformanceEnabled || !isRunningSpan(fragment)) {
```

Comment:
`h`: If `isPerformanceEnabled` is evaluated on each call it could be that it is disabled here and never finish the span. If `isPerformanceEnabled` is set to `false` and we have a running span we could just discard the span.

Reply:
you cannot mutate the `isPerformanceEnabled` after the SDK is inited, so I don't see the point of taking care of this corner case, the same happens on `ActivityLifecycleIntegration`

Reply:
Couldn't you do some crazy stuff like this?
```Java
public class MyApplication extends Application {

  private SentryOptions myOptions;

  @Override
  public void onCreate() {
    super.onCreate();
    
    SentryAndroid.init(
        this,
        options -> {
          myOptions = options;
        });

    // some time passes
    myOptions.setTracesSampleRate(null);
  }
}
```

Reply:
you could, yes, this would be misusing the API hence not really a case that we should take care, if we take this option in consideration, there are N cases that would break the whole SDK, or using reflection and setting non null fields to null etc

Reply:
also `init` should be called only once, its part of the spec.